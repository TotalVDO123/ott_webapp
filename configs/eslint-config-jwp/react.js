module.exports = {
  extends: [
    // Extend the base config
    './typescript',

    // Use recommended React rules
    'plugin:react/recommended',
  ],

  rules: {
    // Not needed in React 17
    'react/react-in-jsx-scope': 'off',
  },

  overrides: [
    {
      files: ['*.jsx', '*.tsx', '*.ts'],
      plugins: [
        // Enable linting React code
        'react',
        'react-hooks',
      ],
      rules: {
        // Help with Hooks syntax
        'react-hooks/rules-of-hooks': 'error',
        'react-hooks/exhaustive-deps': 'error',

        // Handled by Typescript
        'react/prop-types': 'off',

        // This rule causes too many false positives, e.g. with default exports or child render function
        'react/display-name': 'off',
      },
    },
  ],

  settings: {
    react: {
      pragma: 'React',
      version: '17',
    },
  },
  parserOptions: {
    ecmaFeatures: {
      jsx: true,
    },
  },
};
