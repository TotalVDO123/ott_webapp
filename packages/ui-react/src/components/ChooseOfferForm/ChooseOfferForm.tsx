import React from 'react';
import { useTranslation } from 'react-i18next';
import classNames from 'classnames';
import type { FormErrors } from '@jwp/ott-common/types/form';
import type { Offer, ChooseOfferFormData } from '@jwp/ott-common/types/checkout';
import { useConfigStore } from '@jwp/ott-common/src/stores/ConfigStore';
import { testId } from '@jwp/ott-common/src/utils/common';

import Button from '../Button/Button';
import FormFeedback from '../FormFeedback/FormFeedback';
import DialogBackButton from '../DialogBackButton/DialogBackButton';
import LoadingOverlay from '../LoadingOverlay/LoadingOverlay';
import PriceBox from '../PriceBox/PriceBox';

import styles from './ChooseOfferForm.module.scss';

type Props = {
  values: ChooseOfferFormData;
  errors: FormErrors<ChooseOfferFormData>;
  onChange: React.ChangeEventHandler<HTMLInputElement | HTMLTextAreaElement>;
  onSubmit: React.FormEventHandler<HTMLFormElement>;
  onBackButtonClickHandler?: () => void;
  offers: Offer[];
  showOfferTypeSwitch: boolean;
  submitting: boolean;
};

const ChooseOfferForm: React.FC<Props> = ({ values, errors, submitting, offers, showOfferTypeSwitch, onChange, onSubmit, onBackButtonClickHandler }: Props) => {
  const siteName = useConfigStore((s) => s.config.siteName);
  const { t } = useTranslation('account');
  const { selectedOfferType, selectedOfferId } = values;

  return (
    <form onSubmit={onSubmit} data-testid={testId('choose-offer-form')} noValidate>
      {onBackButtonClickHandler ? <DialogBackButton onClick={onBackButtonClickHandler} /> : null}
      <h1 className={styles.title}>{t('choose_offer.title')}</h1>
      <p className={styles.subtitle}>{t('choose_offer.watch_this_on_platform', { siteName })}</p>
      {errors.form ? <FormFeedback variant="error">{errors.form}</FormFeedback> : null}
      {showOfferTypeSwitch && (
        <div className={styles.offerGroupSwitch}>
          <input
            className={styles.radio}
            onChange={onChange}
            type="radio"
            name="selectedOfferType"
            id="svod"
            value="svod"
            checked={selectedOfferType === 'svod'}
          />
          <label className={classNames(styles.label, styles.offerGroupLabel)} htmlFor="svod">
            {t('choose_offer.subscription')}
          </label>
          <input
            className={styles.radio}
            onChange={onChange}
            type="radio"
            name="selectedOfferType"
            id="tvod"
            value="tvod"
            checked={selectedOfferType === 'tvod'}
          />
          <label className={classNames(styles.label, styles.offerGroupLabel)} htmlFor="tvod">
            {t('choose_offer.one_time_only')}
          </label>
        </div>
      )}
      <div className={styles.offers}>
        {!offers.length ? (
          <p>{t('choose_offer.no_pricing_available')}</p>
        ) : (
          offers.map((offer) => <PriceBox key={offer.offerId} offer={offer} selected={selectedOfferId === offer.offerId} onChange={onChange} />)
        )}
      </div>
      {submitting && <LoadingOverlay transparentBackground inline />}
      <Button label={t('choose_offer.continue')} disabled={submitting || !offers.length} variant="contained" color="primary" type="submit" fullWidth />
    </form>
  );
};
export default ChooseOfferForm;
