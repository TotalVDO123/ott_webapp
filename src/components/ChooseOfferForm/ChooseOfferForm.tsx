import React, { useContext } from 'react';
import { useTranslation } from 'react-i18next';

import Button from '../Button/Button';
import CheckCircle from '../../icons/CheckCircle';
import { ConfigContext } from '../../providers/ConfigProvider';
import type { Offer } from '../../../types/checkout';
import FormFeedback from '../FormFeedback/FormFeedback';
import { getOfferPrice } from '../../utils/subscription';
import DialogBackButton from '../DialogBackButton/DialogBackButton';
import type { FormErrors } from '../../../types/form';
import LoadingOverlay from '../LoadingOverlay/LoadingOverlay';
import { IS_DEV_BUILD } from '../../utils/common';

import styles from './ChooseOfferForm.module.scss';

import type { ChooseOfferFormData } from '#types/account';

type Props = {
  values: ChooseOfferFormData;
  errors: FormErrors<ChooseOfferFormData>;
  onChange: React.ChangeEventHandler<HTMLInputElement | HTMLTextAreaElement>;
  onSubmit: React.FormEventHandler<HTMLFormElement>;
  onBackButtonClickHandler?: () => void;
  monthlyOffer?: Offer;
  yearlyOffer?: Offer;
  submitting: boolean;
};

const ChooseOfferForm: React.FC<Props> = ({ values, errors, onChange, onSubmit, submitting, yearlyOffer, monthlyOffer, onBackButtonClickHandler }: Props) => {
  const { siteName } = useContext(ConfigContext);
  const { t } = useTranslation('account');

  const getFreeTrialText = (offer: Offer) => {
    if (offer.freeDays > 0) {
      return t('choose_offer.benefits.first_days_free', { count: offer.freeDays });
    } else if (offer.freePeriods) {
      // t('periods.day')
      // t('periods.week')
      // t('periods.month')
      // t('periods.year')
      const period = t(`periods.${offer.period}`, { count: offer.freePeriods });

      return t('choose_offer.benefits.first_periods_free', { count: offer.freePeriods, period });
    }

    return null;
  };

  return (
    <form onSubmit={onSubmit} data-testid={IS_DEV_BUILD ? 'choose-offer-form' : undefined} noValidate>
      {onBackButtonClickHandler ? <DialogBackButton onClick={onBackButtonClickHandler} /> : null}
      <h2 className={styles.title}>{t('choose_offer.subscription')}</h2>
      <h3 className={styles.subtitle}>{t('choose_offer.all_movies_and_series_of_platform', { siteName })}</h3>
      {errors.form ? <FormFeedback variant="error">{errors.form}</FormFeedback> : null}
      <div className={styles.offers}>
        {monthlyOffer ? (
          <div className={styles.offer}>
            <input
              className={styles.radio}
              onChange={onChange}
              type="radio"
              name="periodicity"
              value="monthly"
              id="monthly"
              checked={values.periodicity === 'monthly'}
              aria-label={t('choose_offer.monthly_subscription')}
            />
            <label className={styles.label} htmlFor="monthly">
              <h4 className={styles.offerTitle}>{t('choose_offer.monthly')}</h4>
              <hr className={styles.offerDivider} />
              <ul className={styles.offerBenefits}>
                {monthlyOffer.freeDays > 0 || monthlyOffer.freePeriods > 0 ? (
                  <li>
                    <CheckCircle /> {getFreeTrialText(monthlyOffer)}
                  </li>
                ) : null}
                <li>
                  <CheckCircle /> {t('choose_offer.benefits.cancel_anytime')}
                </li>
                <li>
                  <CheckCircle /> {t('choose_offer.benefits.watch_on_all_devices')}
                </li>
              </ul>
              <div className={styles.fill} />
              <div className={styles.offerPrice}>
                {getOfferPrice(monthlyOffer)} <small>/{t('periods.month')}</small>
              </div>
            </label>
          </div>
        ) : null}
        {yearlyOffer ? (
          <div className={styles.offer}>
            <input
              className={styles.radio}
              onChange={onChange}
              type="radio"
              name="periodicity"
              value="yearly"
              id="yearly"
              checked={values.periodicity === 'yearly'}
              aria-label={t('choose_offer.yearly_subscription')}
            />
            <label className={styles.label} htmlFor="yearly">
              <h4 className={styles.offerTitle}>{t('choose_offer.yearly')}</h4>
              <hr className={styles.offerDivider} />
              <ul className={styles.offerBenefits}>
                {yearlyOffer.freeDays > 0 || yearlyOffer.freePeriods > 0 ? (
                  <li>
                    <CheckCircle /> {getFreeTrialText(yearlyOffer)}
                  </li>
                ) : null}
                <li>
                  <CheckCircle /> {t('choose_offer.benefits.cancel_anytime')}
                </li>
                <li>
                  <CheckCircle /> {t('choose_offer.benefits.watch_on_all_devices')}
                </li>
              </ul>
              <div className={styles.fill} />
              <div className={styles.offerPrice}>
                {getOfferPrice(yearlyOffer)} <small>/{t('periods.year')}</small>
              </div>
            </label>
          </div>
        ) : null}
      </div>
      {submitting && <LoadingOverlay transparentBackground inline />}
      <Button label={t('choose_offer.continue')} disabled={submitting} variant="contained" color="primary" type="submit" fullWidth />
    </form>
  );
};
export default ChooseOfferForm;
