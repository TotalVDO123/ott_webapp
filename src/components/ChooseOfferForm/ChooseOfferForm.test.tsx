import React from 'react';
import { fireEvent, render } from '@testing-library/react';

import monthlyOffer from '../../fixtures/monthlyOffer.json';
import yearlyOffer from '../../fixtures/yearlyOffer.json';
import type { Offer } from '../../../types/checkout';

import ChooseOfferForm from './ChooseOfferForm';

describe('<OffersForm>', () => {
  test('renders and matches snapshot', () => {
    const { container } = render(
      <ChooseOfferForm
        values={{ periodicity: 'monthly' }}
        errors={{}}
        onChange={vi.fn()}
        onSubmit={vi.fn()}
        submitting={false}
        monthlyOffer={monthlyOffer as Offer}
        yearlyOffer={yearlyOffer as Offer}
      />,
    );

    expect(container).toMatchSnapshot();
  });

  test('checks the monthly offer correctly', () => {
    const { getByLabelText } = render(
      <ChooseOfferForm
        values={{ periodicity: 'monthly' }}
        errors={{}}
        onChange={vi.fn()}
        onSubmit={vi.fn()}
        submitting={false}
        monthlyOffer={monthlyOffer as Offer}
        yearlyOffer={yearlyOffer as Offer}
      />,
    );

    expect(getByLabelText('choose_offer.monthly_subscription')).toBeChecked();
  });

  test('checks the yearly offer correctly', () => {
    const { getByLabelText } = render(
      <ChooseOfferForm
        values={{ periodicity: 'yearly' }}
        errors={{}}
        onChange={vi.fn()}
        onSubmit={vi.fn()}
        submitting={false}
        monthlyOffer={monthlyOffer as Offer}
        yearlyOffer={yearlyOffer as Offer}
      />,
    );

    expect(getByLabelText('choose_offer.yearly_subscription')).toBeChecked();
  });

  test('calls the onChange callback when changing the offer', () => {
    const onChange = vi.fn();
    const { getByLabelText } = render(
      <ChooseOfferForm
        values={{ periodicity: 'monthly' }}
        errors={{}}
        onChange={onChange}
        onSubmit={vi.fn()}
        submitting={false}
        monthlyOffer={monthlyOffer as Offer}
        yearlyOffer={yearlyOffer as Offer}
      />,
    );

    fireEvent.click(getByLabelText('choose_offer.yearly_subscription'));

    expect(onChange).toBeCalled();
  });

  test('calls the onSubmit callback when submitting the form', () => {
    const onSubmit = vi.fn();
    const { getByTestId } = render(
      <ChooseOfferForm
        values={{ periodicity: 'monthly' }}
        errors={{}}
        onChange={vi.fn()}
        onSubmit={onSubmit}
        submitting={false}
        monthlyOffer={monthlyOffer as Offer}
        yearlyOffer={yearlyOffer as Offer}
      />,
    );

    fireEvent.submit(getByTestId('choose-offer-form'));

    expect(onSubmit).toBeCalled();
  });
});
