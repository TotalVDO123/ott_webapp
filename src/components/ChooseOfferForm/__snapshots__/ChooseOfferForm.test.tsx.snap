// Vitest Snapshot v1

exports[`<OffersForm> > renders and matches snapshot 1`] = `
<div>
  <form
    data-testid="choose-offer-form"
    novalidate=""
  >
    <h2
      class="title"
    >
      choose_offer.subscription
    </h2>
    <h3
      class="subtitle"
    >
      choose_offer.all_movies_and_series_of_platform
    </h3>
    <div
      class="offers"
    >
      <div
        class="offer"
      >
        <input
          aria-label="choose_offer.monthly_subscription"
          checked=""
          class="radio"
          id="monthly"
          name="periodicity"
          type="radio"
          value="monthly"
        />
        <label
          class="label"
          for="monthly"
        >
          <h4
            class="offerTitle"
          >
            choose_offer.monthly
          </h4>
          <hr
            class="offerDivider"
          />
          <ul
            class="offerBenefits"
          >
            <li>
              <svg
                aria-hidden="true"
                class="icon"
                focusable="false"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g>
                  <path
                    d="M12 2C6.5 2 2 6.5 2 12S6.5 22 12 22 22 17.5 22 12 17.5 2 12 2M10 17L5 12L6.41 10.59L10 14.17L17.59 6.58L19 8L10 17Z"
                  />
                </g>
              </svg>
               
              choose_offer.benefits.first_periods_free
            </li>
            <li>
              <svg
                aria-hidden="true"
                class="icon"
                focusable="false"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g>
                  <path
                    d="M12 2C6.5 2 2 6.5 2 12S6.5 22 12 22 22 17.5 22 12 17.5 2 12 2M10 17L5 12L6.41 10.59L10 14.17L17.59 6.58L19 8L10 17Z"
                  />
                </g>
              </svg>
               
              choose_offer.benefits.cancel_anytime
            </li>
            <li>
              <svg
                aria-hidden="true"
                class="icon"
                focusable="false"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g>
                  <path
                    d="M12 2C6.5 2 2 6.5 2 12S6.5 22 12 22 22 17.5 22 12 17.5 2 12 2M10 17L5 12L6.41 10.59L10 14.17L17.59 6.58L19 8L10 17Z"
                  />
                </g>
              </svg>
               
              choose_offer.benefits.watch_on_all_devices
            </li>
          </ul>
          <div
            class="fill"
          />
          <div
            class="offerPrice"
          >
            € 6,99
             
            <small>
              /
              periods.month
            </small>
          </div>
        </label>
      </div>
      <div
        class="offer"
      >
        <input
          aria-label="choose_offer.yearly_subscription"
          class="radio"
          id="yearly"
          name="periodicity"
          type="radio"
          value="yearly"
        />
        <label
          class="label"
          for="yearly"
        >
          <h4
            class="offerTitle"
          >
            choose_offer.yearly
          </h4>
          <hr
            class="offerDivider"
          />
          <ul
            class="offerBenefits"
          >
            <li>
              <svg
                aria-hidden="true"
                class="icon"
                focusable="false"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g>
                  <path
                    d="M12 2C6.5 2 2 6.5 2 12S6.5 22 12 22 22 17.5 22 12 17.5 2 12 2M10 17L5 12L6.41 10.59L10 14.17L17.59 6.58L19 8L10 17Z"
                  />
                </g>
              </svg>
               
              choose_offer.benefits.first_days_free
            </li>
            <li>
              <svg
                aria-hidden="true"
                class="icon"
                focusable="false"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g>
                  <path
                    d="M12 2C6.5 2 2 6.5 2 12S6.5 22 12 22 22 17.5 22 12 17.5 2 12 2M10 17L5 12L6.41 10.59L10 14.17L17.59 6.58L19 8L10 17Z"
                  />
                </g>
              </svg>
               
              choose_offer.benefits.cancel_anytime
            </li>
            <li>
              <svg
                aria-hidden="true"
                class="icon"
                focusable="false"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g>
                  <path
                    d="M12 2C6.5 2 2 6.5 2 12S6.5 22 12 22 22 17.5 22 12 17.5 2 12 2M10 17L5 12L6.41 10.59L10 14.17L17.59 6.58L19 8L10 17Z"
                  />
                </g>
              </svg>
               
              choose_offer.benefits.watch_on_all_devices
            </li>
          </ul>
          <div
            class="fill"
          />
          <div
            class="offerPrice"
          >
            € 50,00
             
            <small>
              /
              periods.year
            </small>
          </div>
        </label>
      </div>
    </div>
    <button
      aria-disabled="false"
      class="button primary fullWidth"
      type="submit"
    >
      <span>
        choose_offer.continue
      </span>
    </button>
  </form>
</div>
`;
